lms/djangoapps/courseware/access.py

```diff
@@ -370,17 +396,24 @@ def _can_load_course_overview(user, course_overview):
         The user doesn't have to be enrolled in the course in order to have load
         load access.
     """
-    return (
-        not course_overview.visible_to_staff_only
-        and _can_access_descriptor_with_start_date(user, course_overview, course_overview.id)
-    ) or _has_staff_access_to_descriptor(user, course_overview, course_overview.id)
+    if _has_staff_access_to_descriptor(user, course_overview, course_overview.id):
+        return ACCESS_GRANTED
+
+    if course_overview.visible_to_staff_only:
+        return VisibilityError()
+
+    access_response = _can_access_descriptor_with_start_date(user, course_overview, course_overview.id)
+    if not access_response:
+        return access_response
+
+    return ACCESS_GRANTED
```

Comment:
why not replace lines 405-409 with one statement:

`return _can_access_descriptor_with_start_date(user, course_overview, course_overview.id)`